src/Umbraco.Core/Models/GridAttributesBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Umbraco.Core.Models
+{
+    public abstract class GridAttributesBase : IGridAttributes
+    {
+        /// <inheritdoc />
+        [JsonProperty("config")]
+        public JToken Config { get; set; }
+
+        /// <inheritdoc />
+        [JsonProperty("styles")]
+        public JToken Styles { get; set; }
+
+        /// <inheritdoc />
+        public IEnumerable<KeyValuePair<string, string>> GetAttributes()
```

Comment:
This logic will need to take into account the XSS fixes put in place:  https://github.com/umbraco/Umbraco-CMS/pull/2243/files

Reply:
Currently the GridControl object is converted to a JToken and passed to the view the model.  The grid editor views could be changed to accept a GridControl as the model instead of a dynamic.  With the integration of NestedContent, an alternative would be to not use a separate GridControl object but allow an IPublishedContent instead.  This would require changing the Grid datatype to accept document types.  DocTypeGridEditor sort of allows this, but I'd like to see document types as first class citizens instead.

Reply:
https://github.com/umbraco/Umbraco-CMS/pull/2234/commits/0662f68ce74e6ee1fe0c322794ab01c3f8ba1d9a